client/rpc/client.go

```diff
@@ -77,21 +119,47 @@ func (rc *RpcClient) Close() {
 // registerHandlers registers error and response handles for the rpc client
 func (rs *RpcClient) registerHandlers() {
 
-	rs.nts.RegisterErrorHandler(network.DirectFundRequestMethod, func(data []byte) {
+	rs.nts.RegisterErrorHandler(serde.DirectDefundRequestMethod, func(id uint64, data []byte) {
 		panic(fmt.Sprintf("Objective failed: %v", data))
 	})
-	rs.nts.RegisterResponseHandler(func(data []byte) {
-		rs.nts.Logger.Trace().Msgf("Rpc client received response: %+v", data)
 
-		rpcResponse := serde.JsonRpcDirectFundResponse{}
-		err := json.Unmarshal(data, &rpcResponse)
-		if err != nil {
-			panic("could not unmarshal direct fund objective response")
+	rs.nts.RegisterResponseHandler(func(id uint64, data []byte) {
```

Comment:
Including the id in the function signature makes things easier for the RPC client handling a response. The RPC client can use the id to look up the original request type and then de-serialize `data` to the an exact type based on the original method type.